### ISSUE TITLE
Execution timing of `config` differs from packer when `event` is set.

### ISSUE THREAD

**Issue by gw31415** [tani/vim-jetpack#91]
This init.lua doesn't works properly, while equivalent init.lua works properly in packer.nvim.
```lua
vim.cmd 'packadd vim-jetpack'
require 'jetpack.packer'.startup(function(use)
	use { 'tani/vim-jetpack', opt = 1 }
	use { 'hrsh7th/cmp-cmdline', event = { 'CmdlineEnter' } }
	use {
		'hrsh7th/nvim-cmp',
		event = { 'CmdlineEnter' },
		config = function()
			local cmp = require 'cmp'
			cmp.setup.cmdline(':', {
				mapping = cmp.mapping.preset.cmdline(),
				sources = cmp.config.sources {
					{ name = 'cmdline' },
				}
			})
		end
	}
end)
```

Probably because, in the case of packer.nvim, all plug-ins are installed before all configs are run.

Do you plan to modify it to have the same behavior as packer.nvim?

**Comment by gw31415** [comment]
I created simple docker containers to check this behavior.
[containers.tar.gz](https://github.com/tani/vim-jetpack/files/10095895/containers.tar.gz)
```
containers
├── jetpack
│   ├── Dockerfile
│   └── init.lua
└── packer
    ├── Dockerfile
    └── init.lua
```

**Comment by tani** [comment]
Thank you for reporting the issue. I have confirmed the issue.
Please check solution #92. Cheers,

**Comment by tani** [comment]
I have merged the changes. Please close this issue if you have no problem. Thanks.

**Comment by gw31415** [comment]
Thanks.
I have confirmed that the error that appears on first startup is resolved.

However, I noticed that even after revision, cmp-cmdline is not working even after CmdlineEnter.
It does not seem to be due to the order in which the plugins are loaded (confirmed by print debugging).

Here is vim-jetpack case:
<img width="588" alt="image" src="https://user-images.githubusercontent.com/24710985/204122651-63dcd5de-de7e-4c5b-82d5-ca9714a16249.png">

Here is packer.nvim:case:
<img width="588" alt="image" src="https://user-images.githubusercontent.com/24710985/204122459-61b751b1-d664-482e-b9b4-43510da1c5fb.png">

In the case of vim-jetpack, nvim-cmp is not shown.
These tests is conducted in the containers mentioned in this comment.
> I created simple docker containers to check this behavior. [containers.tar.gz](https://github.com/tani/vim-jetpack/files/10095895/containers.tar.gz)
> 
> ```
> containers
> ├── jetpack
> │   ├── Dockerfile
> │   └── init.lua
> └── packer
>     ├── Dockerfile
>     └── init.lua
> ```

The first problem is solved, so I'm going to close this issue.
Thank you very much.
